### PR TITLE
Add some useful lints to `Cargo.toml`

### PR DESCRIPTION
**Pull request by tcoratger**
Time spent on this PR: 2h

# Pull Request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [ ] Feature
- [X] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Build-related changes
- [ ] Documentation content changes
- [ ] Testing

# What is the new behavior?

According to what we can find here https://github.com/paradigmxyz/reth/blob/main/Cargo.toml, some useful lints are added to `Cargo.toml`. By incorporating these lint settings, we can ensure better adherence to coding standards and practices, leading to cleaner and more maintainable code.

# Does this introduce a breaking change?

- [ ] Yes
- [X] No

### CODE REVIEW

**Review by greged93 — COMMENTED** [review]
I think my big comment on this PR is the amount of useless docs we are adding. In general, it is recommended to add comments/docs ONLY when this brings a real value. IMO comments on functions like
```rust
/// Returns the eoa.
fn eoa(&self) -> Eoa {
   return self.eoa
}
```
feel useless and don't bring much value. They are also very prone to never be updated if we change stuff. If we limit the amount of comments but only keep the very useful and important ones, there is more chance that we will update them if the API changes. 
In general, for comments on functions, I would add comments if the function really does something unclear (although this should be avoided as the identifier of the function should translate enough info to the user), if we do something special in the function (like internal mutability) or if the function can panic.

File: src/eth_provider/database/types/log.rs [hunk 1]
```diff
@@ -4,11 +4,13 @@ use serde::Deserialize;
 /// A transaction receipt as stored in the database
 #[derive(Debug, Deserialize)]
 pub struct StoredLog {
+    /// The log information.
     #[serde(deserialize_with = "crate::eth_provider::database::types::serde::deserialize_intermediate")]
     pub log: Log,
 }
 
 impl From<StoredLog> for Log {
+    /// Converts a [`StoredLog`] into a [`Log`].
```
[the remark this answers is not in the document]
Comment (tcoratger) [reply src/eth_provider/database/types/log.rs]: Yes this is a mandatory doc from the doc lint. Just removed.

File: src/eth_rpc/api/alchemy_api.rs [hunk 2]
```diff
@@ -3,10 +3,12 @@ use jsonrpsee::core::RpcResult as Result;
 use jsonrpsee::proc_macros::rpc;
 use reth_primitives::Address;
 
-// TODO: Define and implement of methods of Alchemy API
+/// TODO: Define and implement of methods of Alchemy API
+/// Represents the Alchemy API.
```
[the remark this answers is not in the document]
Comment (tcoratger) [reply src/eth_rpc/api/alchemy_api.rs]: removed

**Comment by tcoratger** [comment]
> I think my big comment on this PR is the amount of useless docs we are adding. In general, it is recommended to add comments/docs ONLY when this brings a real value. IMO comments on functions like
> 
> ```rust
> /// Returns the eoa.
> fn eoa(&self) -> Eoa {
>    return self.eoa
> }
> ```
> 
> feel useless and don't bring much value. They are also very prone to never be updated if we change stuff. If we limit the amount of comments but only keep the very useful and important ones, there is more chance that we will update them if the API changes. In general, for comments on functions, I would add comments if the function really does something unclear (although this should be avoided as the identifier of the function should translate enough info to the user), if we do something special in the function (like internal mutability) or if the function can panic.

Yes I agree, this is especially done as part of the doc lint which requires documentation for all parts of the code (structure, fields and functions) which are made public. This is mainly an option for publication on crates.io so that all possible contact interfaces with the user are documented. But I agree that sometimes there isn't much to say in the documentation to the point of making it useless. So I deleted this lint for the moment, we will see if we find it useful to add it again in the future.

File: src/eth_provider/error.rs [hunk 3]
```diff
@@ -9,18 +9,31 @@ use thiserror::Error;
 pub enum EthRpcErrorCode {
     /// Custom geth error code, <https://github.com/vapory-legacy/wiki/blob/master/JSON-RPC-Error-Codes-Improvement-Proposal.md>
     Unknown,
+    /// Execution error.
```
[the remark this answers is not in the document]
Comment (tcoratger) [reply src/eth_provider/error.rs]: Fixed

**Review by greged93 — APPROVED** [review]

**Pull request merged**